pkg/trace/api/responses.go

```diff
@@ -90,10 +92,13 @@ func (wc *writeCounter) N() uint64 { return atomic.LoadUint64(&wc.n) }
 // httpRateByService outputs, as a JSON, the recommended sampling rates for all services.
 // It returns the number of bytes written and a boolean specifying whether the write was
 // successful.
-func httpRateByService(w http.ResponseWriter, dynConf *sampler.DynamicConfig) (n uint64, ok bool) {
+func httpRateByService(version uint64, w http.ResponseWriter, dynConf *sampler.DynamicConfig) (n uint64, ok bool) {
 	w.Header().Set("Content-Type", "application/json")
+	currentState := dynConf.RateByService.GetNewState(version) // this is thread-safe
```

Comment:
if there's no updated version, shouldn't you return an empty json?

Reply:
`GetNewState`  returns `nil` for `Rates` and `Mechanisms` if the given version is the same. See [here](https://github.com/DataDog/datadog-agent/pull/10359/files#diff-c346290f514dd944d2568ebb92edb7df9bf88e4f84f5b676da9664ae03d68bf3R89).

Then `omitempty` will remove the fields from the `json`. I'll add `omitempty` on `Rates` as well.